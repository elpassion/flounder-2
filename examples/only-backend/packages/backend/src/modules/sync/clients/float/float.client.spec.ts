import { HttpService } from '@nestjs/axios';
import { FloatClient } from './float.client';
import { withRecording } from '../../../../../test/__tests-support__/tapes/withRecording';

describe('FloatClient', () => {
  describe('getTimeoffs', () => {
    it('throws ClientError when something goes wrong', async () => {
      await withRecording(__dirname, 'bad request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: 'mockToken',
          floatTimeoffTypeId: 472564,
        });
        await expect(() => client.getTimeOffs(new Date('2023-01-04'))).rejects.toThrowError(
          FloatClient.FloatClientError,
        );
      });
    });

    it('returns correct timeoffs', async () => {
      await withRecording(__dirname, 'successful request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: 'mockToken',
          floatTimeoffTypeId: 472564,
        });
        expect(await client.getTimeOffs(new Date('2023-01-04'))).toMatchSnapshot();
      });
    });
  });

  describe('createTimeOff', () => {
    it('throws ClientError when something goes wrong', async () => {
      await withRecording(__dirname, 'bad post request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: 'mockToken',
          floatTimeoffTypeId: 472564,
        });
        await expect(() =>
          client.createTimeOff('2050-01-01', '2020-10-12', 123, 'CALAMARI XXX'),
        ).rejects.toThrowError(FloatClient.FloatClientError);
      });
    });

    it('return created timeoff', async () => {
      await withRecording(__dirname, 'successful post request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: 'mock_token',
          floatTimeoffTypeId: 474348,
        });
        expect(
          await client.createTimeOff('2023-08-15', '2023-08-20', 17948762, 'CALAMARI XXX'),
        ).toMatchSnapshot();
      });
    });
  });

  describe('removeTimeoff', () => {
    it('throws ClientError when something goes wrong', async () => {
      await withRecording(__dirname, 'bad delete request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: '95e558243319b10dAcMEDEDmWVG1NFGvdPlf3yrLBkoC9tFNAu8qpRSr2Y0=',
          floatTimeoffTypeId: 472564,
        });
        await expect(() => client.removeTimeoff(43567890876543456)).rejects.toThrowError(
          FloatClient.FloatClientError,
        );
      });
    });

    it('return delete status', async () => {
      await withRecording(__dirname, 'successful delete request', async () => {
        const client = new FloatClient(new HttpService(), {
          floatToken: '95e558243319b10dAcMEDEDmWVG1NFGvdPlf3yrLBkoC9tFNAu8qpRSr2Y0=',
          floatTimeoffTypeId: 474348,
        });
        expect(await client.removeTimeoff(15095042)).toEqual(204);
      });
    });
  });
});
