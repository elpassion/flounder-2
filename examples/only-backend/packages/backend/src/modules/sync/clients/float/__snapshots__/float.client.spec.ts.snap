// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatClient createTimeOff return created timeoff 1`] = `
Object {
  "created": "2023-01-09 13:09:52",
  "created_by": "741307",
  "end_date": "2023-08-20",
  "full_day": 1,
  "hours": null,
  "modified": "2023-01-09 13:09:52",
  "modified_by": "741307",
  "people_ids": Array [
    17948762,
  ],
  "repeat_end": null,
  "repeat_state": 0,
  "start_date": "2023-08-15",
  "start_time": null,
  "status": 2,
  "status_creator_id": null,
  "status_note": null,
  "timeoff_id": 15095042,
  "timeoff_notes": "CALAMARI XXX",
  "timeoff_type_id": 474348,
  "timeoff_type_name": "Paid Time Off",
}
`;

exports[`FloatClient getTimeoffs returns correct timeoffs 1`] = `
FloatTimeoffResponse {
  "timeoffs": Array [
    FloatTimeoff {
      "calamariId": 0,
      "endDate": 2023-01-16T00:00:00.000Z,
      "id": 15045878,
      "personEmail": undefined,
      "personId": 17946185,
      "serviceType": "FLOAT",
      "startDate": 2023-01-16T00:00:00.000Z,
    },
    FloatTimeoff {
      "calamariId": 666,
      "endDate": 2023-06-01T00:00:00.000Z,
      "id": 15045866,
      "personEmail": undefined,
      "personId": 17946185,
      "serviceType": "FLOAT",
      "startDate": 2023-06-01T00:00:00.000Z,
    },
    FloatTimeoff {
      "calamariId": 123,
      "endDate": 2023-01-02T00:00:00.000Z,
      "id": 15045863,
      "personEmail": undefined,
      "personId": 17946182,
      "serviceType": "FLOAT",
      "startDate": 2023-01-02T00:00:00.000Z,
    },
    FloatTimeoff {
      "calamariId": 46,
      "endDate": 2023-01-10T00:00:00.000Z,
      "id": 15045857,
      "personEmail": undefined,
      "personId": 17946182,
      "serviceType": "FLOAT",
      "startDate": 2023-01-09T00:00:00.000Z,
    },
    FloatTimeoff {
      "calamariId": 456,
      "endDate": 2023-01-05T00:00:00.000Z,
      "id": 15045856,
      "personEmail": undefined,
      "personId": 17946185,
      "serviceType": "FLOAT",
      "startDate": 2023-01-04T00:00:00.000Z,
    },
  ],
}
`;
